update_translations.py

```diff
@@ -0,0 +1,116 @@
+
+javarosa_filename = 'messages_default.txt'
+commcare_filename = 'messages_cc_default.txt'
+ccodk_messages_filename = 'messages_ccodk_default.txt'
+ccodk_strings_filename = 'strings.xml'
+
+javarosa_header_string = '# *** messages_default.txt ***'
+commcare_header_string = '# *** messages_cc_default.txt ***'
+ccodk_messages_header_string = '# *** messages_ccodk_default.txt ***'
+ccodk_strings_header_string = '# *** strings.xml ***'
+
+header_strings = [javarosa_header_string, commcare_header_string, ccodk_messages_header_string, ccodk_strings_header_string]
+
+strings_namespace = '{http://strings_namespace}'
+
+
+def update_translations(new_version_number):
+    new_javarosa_text = get_updated_translations(javarosa_repo, javarosa_subfolder, javarosa_filename)
+    new_commcare_text = get_updated_translations(commcare_repo, commcare_subfolder, commcare_filename)
+    new_ccodk_text = get_updated_translations(commcare_odk_repo, ccodk_messages_subfolder, ccodk_messages_filename)
+    new_strings_text = get_updated_strings_block()
+    new_text_blocks = [new_javarosa_text, new_commcare_text, new_ccodk_text, new_strings_text]
+
+    new_branch_name = checkout_new_translations_branch(new_version_number)
+    backup_old_translations_file()
+    create_updated_translations_file(new_text_blocks)
+    commit_and_push_new_branch(new_version_number, new_branch_name)
+
+
+def checkout_new_translations_branch(new_version_number):
+    chdir_repo(translations_repo)
+    subprocess.call('git checkout master', shell=True)
+    subprocess.call('git pull origin master', shell=True)
+    new_branch_name = '{}_release_additions'.format(new_version_number)
+    subprocess.call('git checkout -b {}'.format(new_branch_name), shell=True)
+    return new_branch_name
+
+
+def backup_old_translations_file():
+    subprocess.call('mv {0} {1}'.format(hq_translations_filename, hq_translations_filename + '.bak'), shell=True)
+
+
+def create_updated_translations_file(new_text_blocks):
+    subprocess.call('touch {}'.format(hq_translations_filename), shell=True)
+    f = open(hq_translations_filename, 'w')
+    for i in range(4):
+        f.write(header_strings[i] + '\n\n')
+        if i < 3:
+            f.write(new_text_blocks[i] + '\n\n')
+        else:
+            f.write(new_text_blocks[i])
+
+
+def commit_and_push_new_branch(new_version_number, new_branch_name):
+    subprocess.call('git add {}'.format(hq_translations_filename), shell=True)
+    subprocess.call("git commit -m '{}'".format('Auto-commit: Update translations file for CommCare release ' + new_version_number),
+                    shell=True)
+    subprocess.call('git push origin {}'.format(new_branch_name), shell=True)
+
+
+def get_updated_translations(repo, relative_path, filename):
+    chdir_repo(repo)
+    subprocess.call('git checkout master', shell=True)
+    os.chdir(relative_path)
+    f = open(filename, 'r')
+    return f.read().strip()
+
+
+def get_updated_strings_block():
```

Comment:
@orangejenny, I was actually with @dannyroberts when I was doing this and asked him the best way to do string building in python, and he said either use join on a list, or something else that I forget. Might very well have been using yield. But since this is never going to actually be a particularly long string, it didn't seem necessary to spend too much time agonizing over optimization.